hadoop-hdds/interface-client/src/main/proto/DatanodeClientProtocol.proto

```diff
@@ -451,7 +451,7 @@ RPC */
 message PutSmallFileRequestProto {
   required PutBlockRequestProto block = 1;
   required ChunkInfo chunkInfo = 2;
-  required bytes data = 3;
+  optional bytes data = 3;
```

Comment:
Data in PutSmallFileRequestProto is null in setupStream

Reply:
I see.  Thanks.

Reply:
is this backward compatible?

Reply:
Yes, changing required to optional is backward compatible since the old code always provides the value and the new code can handle it.

However, it is not forward compatible since the old code may not be able to handle the case that the value is missing.